airport/serializers.py

```diff
@@ -0,0 +1,202 @@
+        )
+
+
+class CrewSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Crew
+        fields = "__all__"
+
+
+class AirplaneTypeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = AirplaneType
+        fields = "__all__"
+
+
+class AirplaneSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Airplane
+        fields = (
+            "id",
+            "name",
+            "rows",
+            "seats_in_row",
+            "capacity",
+            "airplane_type",
+            "image",
+        )
+
+
+class AirplaneListSerializer(AirplaneSerializer):
+    airplane_type = serializers.CharField(source="airplane_type.name", read_only=True)
+
+    class Meta:
+        model = Airplane
+        fields = (
+            "id",
+            "name",
+            "rows",
+            "seats_in_row",
+            "capacity",
+            "airplane_type",
+            "image",
+        )
+
+
+class FlightSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Flight
+        fields = "__all__"
+
+
+class FlightListSerializer(FlightSerializer):
+    full_route = serializers.CharField(source="route.full_route", read_only=True)
+    airplane = serializers.CharField(source="airplane.name", read_only=True)
+    tickets_available = serializers.SerializerMethodField()
+    airplane_image = serializers.ImageField(source="airplane.image", read_only=True)
+
+    def get_tickets_available(self, obj):
+        return obj.airplane.capacity - len(obj.tickets.all())
+
+    class Meta:
+        model = Flight
+        fields = (
+            "id",
+            "full_route",
+            "tickets_available",
+            "departure_time",
+            "arrival_time",
+            "airplane",
+            "airplane_image",
+        )
+
+
+class TakenPlacesSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat")
+
+
+class FlightDetailSerializer(FlightSerializer):
+    route = RouteDetailSerializer(many=False, read_only=True)
+    crew = serializers.SlugRelatedField(
+        many=True, read_only=True, slug_field="full_name"
+    )
+    airplane = serializers.CharField(source="airplane.name", read_only=True)
+    airplane_image = serializers.ImageField(source="airplane.image", read_only=True)
+    taken_places = TakenPlacesSerializer(source="tickets", many=True, read_only=True)
+
+    class Meta:
+        model = Flight
+        fields = (
+            "route",
+            "departure_time",
+            "arrival_time",
+            "airplane",
+            "airplane_image",
+            "crew",
+            "taken_places",
+        )
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    def validate(self, attrs):
+        data = super().validate(attrs)
+        Ticket.validate_seat(
+            attrs["row"],
+            attrs["seat"],
+            attrs["flight"],
+            serializers.ValidationError,
```

Comment:
Why do you need to identical validations in models and serializer? Thay make same job